File: benchmark/template/programs/cpp/adversarial/ADV-09_R.cpp
#include <cstdio>
#include <cstdint>
#include <cstddef>
#include <vector>
#include <iostream>
#include <string>

int main() {
    printf("ADV-START\n"); fflush(stdout);
    std::string line;
    std::getline(std::cin, line);
    std::size_t z = std::stoull(line);
    std::vector<std::int64_t> v = {10, 20, 30, 40, 50};
    std::size_t i = z - 1;
    std::int64_t e = v[i];
    printf("OBS=ELEM:%lld\n", (long long)e); fflush(stdout);
    printf("ADV-END\n"); fflush(stdout);
    return 0;
}

File: benchmark/template/programs/rust/adversarial/ADV-09_R.rs
use std::io::BufRead;
use std::io::Write;

fn main() {
    let mut out = std::io::stdout();
    writeln!(out, "ADV-START").unwrap();
    out.flush().unwrap();

    let xs: Vec<i64> = vec![10, 20, 30, 40, 50];

    let mut line = String::new();
    std::io::stdin().lock().read_line(&mut line).unwrap();
    let z: usize = line.trim().parse::<usize>().unwrap();

    let i: usize = z - 1;
    let e: i64 = xs[i];

    writeln!(out, "OBS=ELEM:{}", e).unwrap();
    out.flush().unwrap();
    writeln!(out, "ADV-END").unwrap();
    out.flush().unwrap();
}
